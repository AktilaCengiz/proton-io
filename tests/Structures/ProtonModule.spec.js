/* eslint-disable no-undef */

const ProtonClient = require("../../src/Structures/ProtonClient");
const ProtonHandler = require("../../src/Structures/ProtonHandler");
const ProtonModule = require("../../src/Structures/ProtonModule");
const checkEqualKeys = require("../checkEqualKeys");

class Client extends ProtonClient {
    constructor() {
        super({
            intents: []
        });

        this.handler = new ProtonHandler(this, {
            directory: ""
        });
    }
}

const client = new Client();

test("properties", () => {
    const mod = new ProtonModule("Mod");
    checkEqualKeys(mod, {
        id: "Mod",
        category: "default",
        filepath: null,
        client: null,
        handler: null
    });
    client.handler.register(mod, null);
    checkEqualKeys(mod, {
        client,
        handler: client.handler
    });
    expect(mod).toMatchSnapshot();
});
