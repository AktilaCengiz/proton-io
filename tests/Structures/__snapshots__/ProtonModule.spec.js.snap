// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properties 1`] = `
ProtonModule {
  "category": "default",
  "client": Object {
    "application": null,
    "channels": Array [],
    "guilds": Array [],
    "owners": Array [],
    "presence": undefined,
    "shard": null,
    "user": null,
    "users": Array [],
  },
  "filepath": null,
  "handler": ProtonHandler {
    "_events": Object {},
    "_eventsCount": 0,
    "_maxListeners": undefined,
    "automateCategories": false,
    "client": Object {
      "application": null,
      "channels": Array [],
      "guilds": Array [],
      "owners": Array [],
      "presence": undefined,
      "shard": null,
      "user": null,
      "users": Array [],
    },
    "directory": "",
    "modules": Array [
      [Circular],
    ],
    Symbol(kCapture): false,
  },
  "id": "Mod",
}
`;
